openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,864 @@
+    // Get the indicators children keys
+    for (var q = 0; q < index.length; q++) {
+        indicatorChildrenKey.push(index[q].name);
     }
 
-    $('#warning-no-layer').dialog({
-        autoOpen: false,
-        hieght: 300,
-        width: 350,
-        modal: true
-    });
-
-    //  New project selection dialog
-    $('#loadProjectDialog').dialog({
-        autoOpen: false,
-        height: 220,
-        width: 350,
-        modal: true
-    });
-
-    // Set custom map div height
-    $('#map').height("300px");
-
-    $('#load-project').button().click(function() {
-        $('#loadProjectDialog').dialog('open');
-    });
-
-    // Remove layer 
-    var removeLayer = function () {
-        var e = document.getElementById('layer-list');
-        var layerId = e.options[e.selectedIndex].value;
-
-        // Look up the layer id using the layer name
-        var layerIdArray = layerNames[layerId];
-        var selectedLayer = layerIdArray.toString();
-
-        // Check in the layer is in the map port
-        if (selectedLayer in layers) {
-            layerControl.removeLayer(layers[selectedLayer]);
-            map.removeLayer(layers[selectedLayer]);
-            delete layers[selectedLayer];
-        }
-        else {
-            showRemoveMsg();
-        }
-    };
-
-    // Get layer names from tilestream
-    var tileStreamLayer = '';
-    var category = '';
-    var selCat = document.getElementById('layer-category');
-    var selLayer = document.getElementById('layer-list');
-
-    // Create a header for the menu drop down
-    var catMenuHeader = document.createElement('option');
-    catMenuHeader.innerHTML = 'Projects:';
-    selCat.appendChild(catMenuHeader);
-
-    $.getJSON(TILESTREAM_API_URL,
-    function(json) {
-
-        var name, cat, type, grids, pDef;
-
-        // Create the category list (build the object)
-        for (var i=0; i < json.length; i++) {
-            name = json[i].mapped_value;
-            cat = json[i].category;
-            type = json[i].type;
-            grids = json[i].grids;
-            pDef = json[i].project_definition;
-
-            if (cat != undefined && type == 'svir-qgis') {
-                categoryList.push(cat);
-                layerNames[name] = [];
-                layersByCat[cat] = [];
-                projectDefinition[name] = [];
-
-                if (grids != undefined) {
-                    var grid = grids.toString();
-                    var gridName = grid.split('/')[4];
-                    layerGrids.push(gridName);
+    // Match each primary indicator with it's respective data value
+    var primaryRiskIndicatorObj = {};
+    for (var ic = 0; ic < la.length; ic++) {
+        var tempObj = la[ic].properties;
+        for (var d = 0; d < indicatorChildrenKey.length; d++) {
+            for(var o in tempObj) {
+                if (indicatorChildrenKey[d] == o) {
+                    var tempValue = indicatorChildrenKey[d];
```

Comment:
Please use some more intuitive names for those temporary variables, so what you are doing here becomes simpler to read (and possibly add some comments)